From: Alexandru Avadanii <Alexandru.Avadanii@enea.com>
Date: Sat, 5 Aug 2017 02:25:38 +0200
Subject: [PATCH] maas: region: database: skip credentials update

FIXME: This should be reverted later, after fixing the MaaS password
update in the salt formula / scripts.

Signed-off-by: Alexandru Avadanii <Alexandru.Avadanii@enea.com>
---
 .../0001-maas-region-skip-credentials-update.patch | 30 ++++++++++++++++++++++
 mcp/patches/patches.list                           |  1 +
 2 files changed, 31 insertions(+)
 create mode 100644 mcp/patches/0001-maas-region-skip-credentials-update.patch

diff --git a/mcp/patches/0001-maas-region-skip-credentials-update.patch b/mcp/patches/0001-maas-region-skip-credentials-update.patch
new file mode 100644
index 0000000..f2104e5
--- /dev/null
+++ b/mcp/patches/0001-maas-region-skip-credentials-update.patch
@@ -0,0 +1,30 @@
+From: Alexandru Avadanii <Alexandru.Avadanii@enea.com>
+Date: Sat, 5 Aug 2017 02:03:01 +0200
+Subject: [PATCH] maas: region: skip credentials update
+
+Password update for maas psql database breaks ulterior acesses
+to maas-region syncdb.
+For now, limit regiond.conf changes to maas_url, and skip
+updating credentials.
+
+Signed-off-by: Alexandru Avadanii <Alexandru.Avadanii@enea.com>
+---
+
+diff --git a/maas/region.sls b/maas/region.sls
+index d3227ca..8a2243d 100644
+--- a/maas/region.sls
++++ b/maas/region.sls
+@@ -6,10 +6,9 @@
+     - names: {{ region.pkgs }}
+
+ /etc/maas/regiond.conf:
+-  file.managed:
+-  - source: salt://maas/files/regiond.conf
+-  - template: jinja
+-  - group: maas
++  file.replace:
++  - pattern: ^maas_url.*$
++  - repl: "maas_url: http://{{ region.bind.host }}/MAAS"
+   - require:
+     - pkg: maas_region_packages
+
diff --git a/mcp/patches/patches.list b/mcp/patches/patches.list
index 0542433..4ee439e 100644
--- a/mcp/patches/patches.list
+++ b/mcp/patches/patches.list
@@ -1 +1,2 @@
+/usr/share/salt-formulas/env: 0001-maas-region-skip-credentials-update.patch
 /usr/share/salt-formulas/env: 0002-opendaylight-formula-neutron.patch
